python/ray/data/_internal/execution/interfaces/execution_options.py

```diff
@@ -23,6 +23,23 @@ class ExecutionResources:
     # Object store memory usage in bytes.
     object_store_memory: Optional[int] = None
 
+    _ZERO = None
+    _INF = None
+
+    @classmethod
+    def zero(cls) -> "ExecutionResources":
+        """Returns an ExecutionResources object with zero resources."""
+        if cls._ZERO is None:
+            cls._ZERO = ExecutionResources(0.0, 0.0, 0)
+        return cls._ZERO
```

Comment:
What's our motivation for caching this?
```suggestion
    @classmethod
    def zero(cls) -> "ExecutionResources":
        """Returns an ExecutionResources object with zero resources."""
        return ExecutionResources(0.0, 0.0, 0)
```

Reply:
One concern I have is that `ExecutionResources` are mutable. 

```python
>>> ExecutionResources.zero().cpu = 1
>>> ExecutionResources.zero()
ExecutionResources(cpu=1, gpu=0.0, object_store_memory=0)
```

This might become a gotcha.

Reply:
good catch. we should make it immutable in the future.